### ISSUE TITLE
Licensing

### ISSUE THREAD

**Issue by gerob311** [PandaScience/FortySecondsCV#49]
Hey, I think this is a great CV template! I'm a bit lost on the best way to comply with the BSD license though. Clause 2 says
```
2. Redistributions in binary form must reproduce the above copyright notice,
this list of conditions and the following disclaimer in the documentation
and/or other materials provided with the distribution.
```

I notice that the sample does not put the licence and copyright notice in the generated document. Is the expectation that the text would be included in a cover letter or email when providing a resume to a potential employer? I suspect an employer wouldn't be expecting that, and it might not be a great look. I'm just wondering, is this an unintended consequence of using the BSD 3-Clause licence?

**Comment by PandaScience** [comment]
Hello there! 

Nice catch! No expectation to put anything in any compiled pdf :wink: . Totally agree that it would look weird and is absolutely unintended.

Back when I created the repo I probably just took any "permissive" license without thinking too much about it. Could've been MIT as well. I wasn't fully aware of the "binary issue" in BSD for LaTeX, but I remember that I didn't want to go with LPPL (which seems to be kind of a default in the LaTeX / CTAN world) because it felt too restrictive. On the other hand, Creative Commons seemed like a bad choice b/c it is intended for documents, not source code.

Fun fact: all my _compiled_ lecture pdfs are indeed licensed under CC BY-SA 4.0 :sweat_smile: 

So long story short, it's probably a good idea to re-license this project. In case you have more insights or experience with LaTeX class licensing, what would you prefer? I really don't care too much.  From my limited legal understanding, [MIT](https://choosealicense.com/licenses/mit/) should solve at least the "binary issue"; [ISC](https://choosealicense.com/licenses/isc/) seems to be the currently preferred one by the BSD people. :man_shrugging: 

Btw, is [this](https://opensource.stackexchange.com/questions/15130/how-does-the-3-clause-bsd-license-apply-to-latex-and-pdfs) by any chance your thread?  :sweat_smile:

**Comment by gerob311** [comment]
Sorry to take so long to get back to you. I wanted to have a good think about it, and things got rather busy for me lately.

Yes, that is my StackExchange post. As you can see, I'm hardly an expert on open source licenses. But I've been talking this over with a friend, and I can offer some thoughts.

The problem I see with the ISC license, is that on the page you linked, it says that it is functionally equivalent to the BSD 2-Clause licence, which still has that binary clause. ISC and MIT don't explicitly mention a binary, but they talk about software, and I think that includes a compiled form.

I've been looking at some licences on [this page](https://spdx.org/licenses/). There's a few you might find interesting.
The [BSD 1-Clause License](https://spdx.org/licenses/BSD-1-Clause.html)only retains the first clause about the source code. [BSD Source Code Attribution](https://spdx.org/licenses/BSD-Source-Code.html) is similar, but also retains the clause about not using your name to promote other things.

The [BSL 1.0](https://spdx.org/licenses/BSL-1.0.html) might not be the most applicable, but I thought it was interesting because it explicitly states that the notice doesn't need to be distributed with compiled forms of the software. 

Finally, I'll point out that there seems to be some precedent for using a common license, and then including exceptions to the licence. Here's an [example](https://spdx.org/licenses/Texinfo-exception.html).

So as you can tell from my StackExchange post, I'm hoping you'll pick something that'll still let me port some of the features you've created to a different language, while retaining the license. But of course, it's up to you what you want to require!

**Comment by PandaScience** [comment]
Sorry for the late reply, I've also had some quite busy weeks. 

Generally, as I mentioned before, I'm open for any permissive license as long as the general direction leans towards "more open" not "more restrictive". From your selection (thx for the research, btw :pray: ) I like the [BSD Source Code Attribution](https://spdx.org/licenses/BSD-Source-Code.html) the best, although you cannot find it anywhere else except on SPDX  - but it at least has an ID. :sweat_smile: . 
It's basically BSD 3-Clause without the 2nd condition.

Regarding the functional equivalency (i.e. compatibility :grey_question:): That's in fact what I was looking for because I'm not entirely sure how a change in license of an open-source project with multiple (external) contributors and without an explicit copyright assignment works. My initial thought was that maybe with an equivalent license, I could save the hassle of involving the contributors. But tbh that feels wrong, even if the submitted PRs make only a tiny fraction of the code. 

(sigh...annoying legal stuff... :roll_eyes: :pensive:)

The GitHub [terms of service](https://github.com/github/site-policy/blob/1263bf0164cfe60d0900e70ba9126855c7c23c6c/Policies/github-terms-of-service.md#6-contributions-under-repository-license) state:

> Whenever you make a contribution to a repository containing notice of a license, you license your contribution under the same terms, and you agree that you have the right to license your contribution under those terms. If you have a separate agreement to license your contributions under different terms, such as a contributor license agreement, that agreement will supersede.

So technically, I assume, we would have to ask and wait for each and every contributor to agree to a change of license. Or remove all contributions, re-license, and ask to re-submit under the new terms - which feels absurd and I also don't have the time for.  I'm also uncertain in which form the written consent must be provided, but will just assume that comments in this issue will be sufficient although they could technically be edited but the history will be retained anyhow. :shrug:

[Wikipedia](https://en.wikipedia.org/wiki/Software_relicensing) states on this issue:

> While in the [free and open-source](https://en.wikipedia.org/wiki/Free_and_open-source) domain achieving 100% coverage of all authors is often impossible due to the many contributors involved, often it is assumed that a great majority is sufficient. For instance, [Mozilla](https://en.wikipedia.org/wiki/Mozilla) assumed an author coverage of 95% to be sufficient.[[4]](https://en.wikipedia.org/wiki/Software_relicensing#cite_note-4) Others in the free and open-source software (FOSS) domain, such as [Eric S. Raymond](https://en.wikipedia.org/wiki/Eric_S._Raymond), came to different conclusions regarding the requirements for relicensing of a whole code base.[[5]](https://en.wikipedia.org/wiki/Software_relicensing#cite_note-5)

With reference [4] and [5] containing basically:

> [4]: It would suffice if there was permission from the copyright holders of 95% of the source code and no objections from the holders of the other 5%.

> [5]: You can change the license on a piece of code under any of the following conditions: If you are the sole copyright holder[...]If you are the sole registered copyright holder[...] If you obtain the consent of all other copyright holders[...]If no other copyright holder could be harmed by the change

One could argue that by the latter, more restrictive comment, I could just make the change on my own since 

* I don't see a chance that any of the contributors could be "harmed"
* I think no one really expected anyone to include the copyright notice in a compiled PDF

In case you don't see any legal chance that allows me to just make that change, I would propose the following:

1. Create a PR with the adapted license.
2. Ask the other 8 contributors to clearly state their consent there in form of a traceable issue comment.
3. With all approvals obtained, the PR will be merged. 

Of course there is a chance that not all will reply or can be reached. 

WDYT?

(wow...what an amount of work for such an insignificant side-project :face_exhaling: )

**Comment by PandaScience** [comment]
Well, it took almost a year but eventually, we're done with this issue 🥳.
Thanks again for the hint and discussion 👍.
